clusterloader2/pkg/framework/client/objects.go

```diff
@@ -99,6 +102,37 @@ func DeleteNamespace(c clientset.Interface, namespace string) error {
 	return RetryWithExponentialBackOff(retryFunction(deleteFunc, apierrs.IsNotFound))
 }
 
+// ListNamespaces returns list of existing namespace names.
+func ListNamespaces(c clientset.Interface) ([]string, error) {
```

Comment:
It doesn't really seem worthwhile to have this extra function. Could you move the List() call into `ListAutomanagedNamespaces` (only place where it's called)?

Reply:
I want to keep it as it is.
`client/objects.go` is responsible for cluster operation, while `framework.go` is a facade that makes this generic operation to be clusterloader specific. Therefore, `client/objects.go` returns list of all namespaces, while `framework.go` only filters out non-autogenerated.

Reply:
I'm not saying about how it is used. What I mean is this function is pretty much just a `c.CoreV1().Namespaces().List()` call which you can directly do inside any function where you need the namespaces.

Reply:
I understand your point. However, this change would break the convention I described in previous comment.

Reply:
I've just realized that this implementation was missing retries. I've added it.